sympy/core/add.py

```diff
@@ -594,10 +594,12 @@ def _eval_is_positive(self):
     def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
-            c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
+            c, a = self.as_independent(*self.free_symbols)
```

Comment:
@smichr problem in this ```as_independent(*self.free_symbols)``` function . that's return sometime wrong result.sometime return ```correct``` ans  for same input. try this example ::
```
>>> x = Symbol('x', integer=True, negative=True)
>>> (1 - 1/x).is_nonpositive
```
it's return ```None```
```
>>> x = Symbol('x', integer=True, negative=True)
>>> (1 - 1/x).is_nonnegative
True
>>> (1 - 1/x).is_nonpositive
False
```
that's return ```correct``` ans for same condition.